power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -238,6 +242,8 @@ struct MapArrayVisitor : DefaultErrorVisitor<MapArrayVisitor<map_array>> {
         assert(size == 0);
         return true;
     }
+
+    MapArrayVisitor() = default;
```

Comment:
What is the clang-tidy complaint here?

Reply:
A public constructor is required. Since we now use CRTP with a private constructor, we need to expose it using a defaulted constructor